priv/filelock.go

```diff
@@ -20,70 +20,40 @@
 package priv
 
 import (
-	"errors"
-	"os"
-	"syscall"
+	sys "syscall" // XXX: if we are only targeting 15.04+, i.e. golang 1.5+, we should move to golang.org/x/sys
 )
 
-// ErrNotLocked is returned when an attempts is made to unlock an
-// unlocked FileLock.
-var ErrNotLocked = errors.New("not locked")
-
-// FileLock is a Lock file object used to serialise access for
-// privileged operations.
-type FileLock struct {
-	Filename string
-	realFile *os.File
-}
-
-// NewFileLock creates a new lock object (but does not lock it).
-func NewFileLock(path string) *FileLock {
-	return &FileLock{Filename: path}
-}
-
-// Lock the FileLock object.
-// Returns ErrAlreadyLocked if an existing lock is in place.
-func (l *FileLock) Lock(blocking bool) error {
-	// XXX: don't try to create exclusively - we care if the file failed to
-	// be created, but we don't care if it already existed as the lock
-	// _on_ the file is the most important thing.
-	flags := (os.O_CREATE | os.O_WRONLY)
-
-	f, err := os.OpenFile(l.Filename, flags, 0600)
+// LockedFile is a handle you can use to Unlock a file lock.
+type LockedFile int
+
+// FileLock opens (and possibly creates) a new file at the given path
+// and applies an exclusive advisory lock on it.
+//
+// If the file already has an advisory lock, the `blocking` flag
+// determins whether to block until it is removed or return
+// ErrAlreadyLocked.
+func FileLock(path string, blocking bool) (LockedFile, error) {
```

Comment:
just wrote it out, and LockFile vs FileLock is a wash; if to you it reads better as LockFile, I'll swap them.

"blocking" vs "block", to me blocking wins by a mile; "block" just feels wrong; it gets me thinking, again, of the noun and not the verb, as in a crypto block.

I'm trying to think of other places where you'd use a verb as a parameter, and not finding any, so this might also be why it feels wrong.